react-relay/index.d.ts

```diff
@@ -31,6 +31,24 @@ declare module "react-relay" {
         response: any
     }
 
+    type RelayMutationStatus =
+      'UNCOMMITTED' | // Transaction hasn't yet been sent to the server. Transaction can be committed or rolled back.
+      'COMMIT_QUEUED' | // Transaction was committed but another transaction with the same collision key is pending, so the transaction has been queued to send to the server.
+      'COLLISION_COMMIT_FAILED' | //Transaction was queued for commit but another transaction with the same collision key failed. All transactions in the collision queue, including this one, have been failed. Transaction can be recommitted or rolled back.
+      'COMMITTING' | // Transaction is waiting for the server to respond.
+      'COMMIT_FAILED';
+
+    class RelayMutationTransaction {
+      applyOptimistic(): RelayMutationTransaction;
+      commit(): RelayMutationTransaction;
+      recommit(): void;
+      rollback(): void;
+      getError(): Error;
+      getStatus(): RelayMutationStatus;
+      getHash(): string;
+      getID(): string;
```

Comment:
While it is true that [`ClientMutationID` is a `string`](https://github.com/facebook/relay/blob/9e1a1309420f684aab49cdbe98074094236feab2/src/tools/RelayInternalTypes.js#L71), I wonder what the convention is in DefinitelyTyped, should we add `type ClientMutationID = string;`?